src/jit/importer.cpp

```diff
@@ -3895,14 +3923,25 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
     const char* namespaceName = nullptr;
     const char* methodName    = info.compCompHnd->getMethodNameFromMetadata(method, &className, &namespaceName);
 
-    if ((namespaceName != nullptr) && strcmp(namespaceName, "System") == 0)
+    // Currently we only have intrinsics at the method level, so we can check that
+    // namespaceName, className, and methodName are all not null upfront.
+
+    if ((namespaceName != nullptr) && (className != nullptr) && (methodName != nullptr))
```

Comment:
Nit: It would be nice to early out here to save an extra level of indention for the actual matching code that is likely going to grow pretty big.